### PR TITLE
Fix nano.Ts.StringFloat() for negative time

### PR DESCRIPTION
**Pull request by mattnibs**

### CODE REVIEW

**Comment by mattnibs** [comment]
> This doesn't fix the bug...

Hmm, what zq command and against what dataset where you running to get this conclusion?

**Comment by mattnibs** [comment]
>  I think the ns<0 check should be done in ts.Split()

Idk, I think returning both numbers as negative in Split() is correct. That way you can always do sec * 1e9 + ns to rebuild the int64

**Comment by henridf** [comment]
@mattnibs try running on a zeek log with a negative timestamp and you'll see something like `-1418429426.-887384` in the output. The problem is that not all writers use ts.StringFloat. 

For example zeekio calls record.ZeekStrings which uses Split.

**Comment by henridf** [comment]
```sh
$ cat ntp-neg.log 
#separator \x09
#set_separator  ,
#empty_field    (empty)
#unset_field    -
#path   ntp
#open   2019-11-08-11-44-17
#fields ts      uid     id.orig_h       id.orig_p       id.resp_h       id.resp_p       version mode    stratum poll    precision       root_delay      root_disp       ref_id  ref_time        org_time  rec_time        xmt_time        num_exts
#types  time    string  addr    port    addr    port    count   count   count   interval        interval        interval        interval        string  time    time    time    time    count
1521911732.113288       CdnlQx2eOy4ceCIp2d      10.47.1.30      36352   85.199.214.99   123     4       3       0       1024.000000     4.294967e+09    0.000000        0.000000        \x00\x00\x00\x00  0.000000        0.000000        0.000000        -1418429426.887384      0

22:01 ~/work/looky/zq-sample-data/zeek-default(master)
$ cat ntp-neg.log | zq -f zeek -
#separator \x09
#set_separator  ,
#empty_field    (empty)
#unset_field    -
#path   ntp
#fields ts      uid     id.orig_h       id.orig_p       id.resp_h       id.resp_p       version mode    stratum poll    precision       root_delay      root_disp       ref_id  ref_time        org_time  rec_time        xmt_time        num_exts
#types  time    string  addr    port    addr    port    count   count   count   interval        interval        interval        interval        string  time    time    time    time    count
1521911732.113288       CdnlQx2eOy4ceCIp2d      10.47.1.30      36352   85.199.214.99   123     4       3       0       1024    4294967000      0       0       \x00\x00\x00\x00        0.000000 0.000000 0.000000        -1418429426.-887384     0
```

**Comment by mattnibs** [comment]
@henridf thanks!

**Comment by mattnibs** [comment]
@henridf ready for another look

File: pkg/zng/record.go [hunk 1]
```diff
@@ -183,16 +183,11 @@ func (r *Record) ZeekStrings(precision int, utf8 bool) ([]string, bool, error) {
 			if err != nil {
 				return nil, false, err
 			}
-			sec, ns := ts.Split()
-			if precision == 6 && isHighPrecision(ns) {
+			if precision == 6 && isHighPrecision(ts) {
```
Comment (henridf) [inline pkg/zng/record.go]: removing the use of Split() here makes sense... ZeekStrings shouldn't  be fiddling around with that

**Pull request merged**